wondrous-app/components/BoardLock/index.tsx

```diff
@@ -0,0 +1,65 @@
+import { useOrgBoard, usePodBoard, useCreateEntityContext } from 'utils/hooks';
+import { PRIVACY_LEVEL, PERMISSIONS } from 'utils/constants';
+import { useMe } from 'components/Auth/withAuth';
+import { BoardLockWrapper, BoardOverlay, OverlayPopup, OverlayPopupTitle } from './styles';
+import SkeletonBoard from 'components/Common/SkeletonBoard';
+import { HeaderButton } from 'components/organization/wrapper/styles';
+import { useRouter } from 'next/router';
+
+const BoardLock = ({ children, handleJoinClick, requestSent }) => {
+  const orgBoard = useOrgBoard();
+  const podBoard = usePodBoard();
+  const board = orgBoard || podBoard;
+  const entityContext = useCreateEntityContext();
+  const router = useRouter();
+  const user = useMe();
+  if (!board) return children;
+  const { userOrgs } = entityContext;
+
+  const permissionsWithAccess = [
+    PERMISSIONS.FULL_ACCESS,
+    PERMISSIONS.CREATE_TASK,
+    PERMISSIONS.EDIT_TASK,
+    PERMISSIONS.REVIEW_TASK,
+  ];
+
+  const hasNoPodAccess =
+    board?.pod?.privacyLevel === PRIVACY_LEVEL.private &&
+    !board?.userPermissionsContext?.podPermissions[board?.podId]?.some((el) => permissionsWithAccess.includes(el));
+
+  const isNotAMemberOfTheOrg =
+    !userOrgs || !userOrgs?.getUserOrgs?.find((org) => org.id === board?.orgId) || hasNoPodAccess;
```

Comment:
better to use `some` instead of `find`. 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some

`userOrgs?.getUserOrgs?.some((org) => org.id === board?.orgId`

Reply:
true, done